python/prophet/forecaster.py

```diff
@@ -1198,31 +1205,56 @@ def predict(self, df=None):
             raise Exception('Model has not been fit.')
 
         if df is None:
-            df = self.history.copy()
+            if periods is not None:
+                df = self.setup_dataframe(
+                    self.make_future_dataframe(
+                        periods=int(periods),
+                        freq=freq,
+                        include_history=include_history,
+                    )
+                )
+            else:
+                df = self.history.copy()
         else:
             if df.shape[0] == 0:
                 raise ValueError('Dataframe has no rows.')
             df = self.setup_dataframe(df.copy())
 
         df['trend'] = self.predict_trend(df)
-        seasonal_components = self.predict_seasonal_components(df)
-        if self.uncertainty_samples:
-            intervals = self.predict_uncertainty(df)
-        else:
-            intervals = None
 
-        # Drop columns except ds, cap, floor, and trend
-        cols = ['ds', 'trend']
+        # 'floor' col is required even if not using logistic, for sample_model with mcmc
+        extra_cols = ['t']
+        cols = ['ds', 'trend', 't', 'floor']
         if 'cap' in df:
             cols.append('cap')
-        if self.logistic_floor:
-            cols.append('floor')
+        if not self.logistic_floor:
+            extra_cols.append('floor')
+
         # Add in forecast components
-        df2 = pd.concat((df[cols], intervals, seasonal_components), axis=1)
+        df2 = pd.concat(
+            (df[cols], self.predict_seasonal_components(df)),
+            axis=1
+        )
         df2['yhat'] = (
-                df2['trend'] * (1 + df2['multiplicative_terms'])
-                + df2['additive_terms']
+                df2['trend'] * (1 + df2['multiplicative_terms'])  # noqa
+                + df2['additive_terms']  # noqa
         )
+        # some shenanigans are required to get column order same as previous versions
```

Comment:
Any specific reason it is important to keep the exact same column order?

Reply:
My reasoning was that some users might be trying to pull yhat by using forecast.iloc[:, -1] or something similar, and I don't want to break their code if I can avoid it.